crates/oxc_linter/src/rules/react/checked_requires_onchange_or_readonly.rs

```diff
@@ -0,0 +1,294 @@
+    /// It also warns when checked and defaultChecked properties are used together.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <input type="checkbox" checked />
+    /// <input type="checkbox" checked defaultChecked />
+    /// <input type="radio" checked defaultChecked />
+    ///
+    /// React.createElement('input', { checked: false });
+    /// React.createElement('input', { type: 'checkbox', checked: true });
+    /// React.createElement('input', { type: 'checkbox', checked: true, defaultChecked: true });
+    ///
+    /// // Good
+    /// <input type="checkbox" checked onChange={() => {}} />
+    /// <input type="checkbox" checked readOnly />
+    /// <input type="checkbox" checked onChange readOnly />
+    /// <input type="checkbox" defaultChecked />
+    ///
+    /// React.createElement('input', { type: 'checkbox', checked: true, onChange() {} });
+    /// React.createElement('input', { type: 'checkbox', checked: true, readOnly: true });
+    /// React.createElement('input', { type: 'checkbox', checked: true, onChange() {}, readOnly: true });
+    /// React.createElement('input', { type: 'checkbox', defaultChecked: true });
+    /// ```
+    CheckedRequiresOnchangeOrReadonly,
+    correctness
+);
+
+impl Rule for CheckedRequiresOnchangeOrReadonly {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::JSXOpeningElement(jsx_opening_el) => {
+                let Some(element_type) = get_element_type(ctx, jsx_opening_el) else { return };
+                if element_type != "input" {
+                    return;
+                }
+
+                let (checked_span, default_checked_span, is_missing_property) =
+                    jsx_opening_el.attributes.iter().fold(
+                        (None, None, true),
+                        |(checked_span, default_checked_span, is_missing_property), attr| {
+                            if let JSXAttributeItem::Attribute(jsx_attr) = attr {
+                                let name = get_jsx_attribute_name(&jsx_attr.name);
+                                (
+                                    if name == "checked" {
+                                        Some(jsx_attr.span)
+                                    } else {
+                                        checked_span
+                                    },
+                                    if default_checked_span.is_none() && name == "defaultChecked" {
+                                        Some(jsx_attr.span)
+                                    } else {
+                                        default_checked_span
+                                    },
+                                    is_missing_property
+                                        && !(name.contains("onChange")
+                                            || name.contains("readOnly")),
```

Comment:
Looks we should use `==`, consider we have a attribute and it naming `onChange2` or `readOnlyXX`

Reply:
Thank you!
I've corrected to use `==`.
5825a98feca9e1a84a05bfc4e2148f4822e4e692